### ISSUE TITLE
Tutorial yields "tensorboard: Command not found"

### ISSUE THREAD

**Issue by drscotthawley** [tensorflow/tensorflow#2115]
GitHub issues are for bugs / installation problems / feature requests.  
For general support from the community, see [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow).
To make bugs and feature requests more easy to find and organize, we close issues that are deemed
out of scope for GitHub Issues and point people to StackOverflow.

For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

### Environment info
Operating System: Mac OS X Yosemite, 10.10.5
MacPorts
python 2.7

Installed version of CUDA and cuDNN:  None
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):

If installed from binary pip package, provide:

1. Which pip package you installed.
 % sudo pip install --upgrade https://storage.googleapis.com/tensorflow/mac/tensorflow-0.8.0-py2-none-any.whl
Collecting tensorflow==0.8.0 from https://storage.googleapis.com/tensorflow/mac/tensorflow-0.8.0-py2-none-any.whl
  Downloading https://storage.googleapis.com/tensorflow/mac/tensorflow-0.8.0-py2-none-any.whl (19.3MB)
    100% |████████████████████████████████| 19.3MB 53kB/s 
Requirement already up-to-date: six>=1.10.0 in /opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages (from tensorflow==0.8.0)
Collecting protobuf==3.0.0b2 (from tensorflow==0.8.0)
  Downloading protobuf-3.0.0b2-py2.py3-none-any.whl (326kB)
    100% |████████████████████████████████| 327kB 2.0MB/s 
Collecting wheel (from tensorflow==0.8.0)
  Downloading wheel-0.29.0-py2.py3-none-any.whl (66kB)
    100% |████████████████████████████████| 71kB 10.1MB/s 
Collecting numpy>=1.10.1 (from tensorflow==0.8.0)
  Downloading numpy-1.11.0-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (3.9MB)
    100% |████████████████████████████████| 3.9MB 265kB/s 
Collecting setuptools (from protobuf==3.0.0b2->tensorflow==0.8.0)
  Downloading setuptools-20.10.1-py2.py3-none-any.whl (509kB)
    100% |████████████████████████████████| 512kB 1.9MB/s 
Installing collected packages: setuptools, protobuf, wheel, numpy, tensorflow
  Found existing installation: setuptools 19.2
    Uninstalling setuptools-19.2:
      Successfully uninstalled setuptools-19.2
  Found existing installation: numpy 1.10.4
    DEPRECATION: Uninstalling a distutils installed project (numpy) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
    Uninstalling numpy-1.10.4:
      Successfully uninstalled numpy-1.10.4
Successfully installed numpy-1.11.0 protobuf-3.0.0b2 setuptools-20.10.1 tensorflow-0.8.0 wheel-0.29.0

2. The output from python -c "import tensorflow; print(tensorflow.__version__)".
% python -c "import tensorflow; print(tensorflow.__version__)"
0.8.0

If installed from sources, provide the commit hash:

### Steps to reproduce
1: Read  the documentation at https://www.tensorflow.org/versions/r0.8/how_tos/summaries_and_tensorboard/index.html:
"Launching TensorBoard

To run TensorBoard, use the command

tensorboard --logdir=path/to/log-directory"

2: Do what it says
```
% tensorboard --logdir=path/to/log-directory
tensorboard: Command not found.
```

### What have you tried?
1: Try rehash
```
% rehash
% tensorboard --logdir=path/to/log-directory
tensorboard: Command not found.
```
2:
```
% locate tensorboard
%
```

3:

```
% which tensorboard
tensorboard: Command not found.

```

4:

Check Google, find StackExchange post "How to install tensorboard"
http://stackoverflow.com/questions/33634008/how-to-install-tensorboard
Comments suggest that if one installs via pip (as I did), then tensorboard is available on the command-line, but it is not.
Alternate invocation of tensorboard via python is not the same as in the Tutorial but does work....
```
% python /opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow/tensorboard/tensorboard.py --logdir=/tmp/mnist_logs
Starting TensorBoard 16 on port 6006
(You can navigate to http://0.0.0.0:6006)...
``` 
and everything works.

...Conclusion is that either online Tutorial documentation is incorrect and needs revision, or pip installation does not perform as advertised, or...user error.

Workarounds:

1.
```
% alias tensorboard 'python /opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow/tensorboard/tensorboard.py'
% tensorboard --logdir=/tmp/mnist_logs
```
2.
```
`echo '#! /usr/bin/env python' > tmp.txt ; cat tmp.txt /opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow/tensorboard/tensorboard.py > tensorboard ; chmod u+x tensorboard ; rm -f tmp.txt; sudo mv tensorboard /opt/local/bin/
```

### Logs or other output that would be helpful
(If logs are large, please upload as attachment).

**Comment by poxvoculi** [comment]
Thanks for the detailed report.  It sounds like the pip installation didn't work for you for reasons that we can't reproduce.  Would you consider the virtualenv install?  Under that install you should end up with a small executable 'tensorboard' in the bin directory under the virtualenv directory.    Its contents should be

#!<path to virtualenv directory>/bin/python

# -*- coding: utf-8 -*-
import re
import sys

from tensorflow.tensorboard.tensorboard import main

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())

**Comment by drscotthawley** [comment]
Yes, after virtualenv install I find a "tensorboard" file in the bin directory.  Contents are just as you say.

Figured out the problem:
Uninstalled & re-installed TensorFlow in my main Mac environment, and I learned something: As a MacPorts user, I'm used to running things from out of the path /opt/local/bin.  When you install a python package via MacPorts, that's where the executables go --- even if they're just symbolic links to files to a main python repository in /opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin/

pip installs things into the latter directory, but apparently does NOT add the symbolic link to /opt/local/bin 

This has never been an issue (or even come up) for me before, because I've only installed (non-executable) packages which load from within python.
In conclusion, there *is* a /opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin/tensorboard

This is a pip /  MacPorts-SOP mismatch / user error*, and nothing to do with tensorboard in particular.  
 Please close this issue.   Thanks for your help.



*my 'locate' database was in the process of updating but hadn't completed

**Comment by ShivangiM** [comment]
I faced same issue, and when I typed 
`which tensorboard`
it returned nothing, and following solved my problem
`python -m tensorflow.tensorboard --logdir=path/to/log-directory`

**Comment by saggzz** [comment]
1. make sure that you not trying to run tensorboard global, while your tensorflow is installed in a virtualenv (pycharm egg). Install tensorflow with some command line (pip install tensorflow) globally. 
2. sometimes port 6006 is not working while you using windows, i guess cause it is identified as "Ports normally used by Trojans" (WinHole port 808 and BadBlood port 6006).
Instead ouf 6006 try this: 
tensorboard --logdir=PATH --host localhost --port 8088

**Comment by Red-Eyed** [comment]
I faced with same issue. Ubuntu 19.04 tensorflow 1.13  
`python -m tensorboard.main --logdir=logdir`

**Comment by maxmax1992** [comment]
add the directory of the `which tensorboard` (not including the `/tensorboard`) to the `$PATH` env variable.
in zsh shell example:
`which tensorboard`
```
> ~/.local/bin/tensorboard
```
this command prepends the tensorboard path to the `$PATH` env variable
`export PATH=$HOME/.local/bin:$PATH`

**Comment by YuseqYaseq** [comment]
> 
> 
> I faced with same issue. Ubuntu 19.04 tensorflow 1.13
> This works: `python -m tensorboard.main --logdir=logdir`

Also worked for me on Windows 10.

**Comment by gargivaidya** [comment]
Quickest solution - 

`echo "alias tensorboard='python3 -m tensorboard.main'" >> ~/.bash_profile`

After adding this to your .bash_profile you can use 

`tensorboard --logdir=/path`
